### PR TITLE
Feature/39521

### PR DESCRIPTION
**Pull request by manishdixit78**
Define .env file with following properties:
PORT=9000
NODE_ENV=dev

Create a folder src
Create a file Server.ts inside src
Define a class with the given methods and properties in the ticket#39521
Export Server class.

Create a folder config inside src
- Create a file IConfig.ts which contains interface of config (properties like port and env)
- Create a file configuration.ts
Use dotenv module for extracting config from .env file.
Define config using interface(IConfig).
Use Object.freeze for protecting config object against any modifications after the declaration.
Export config

Create a file index.ts
Import config and Server.
Create an instance of Server by passing config in constructor
Call methods bootstrap and run using instance
App should be running and success message.

### CODE REVIEW

File: extra/index.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+// import the hasPermission from utils/index and permissions from constants
+import {hasPermission} from './utils/index.js';
+import {permissions} from './constants.js';
+
+// calling the hasPermission method and pass the arguments.
+hasPermission(permissions.getUsers,'head-trainer','all');
+console.log();
+
+// import the functions from validateUsers and validdateEmail from util/index
+import {validateUsers} from './utils/index';
```
Comment (faiyaz24) [inline extra/index.js:65]: Make sure import should be on of the file.

File: extraTs/constants.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    delete: [],
+    },
+    'getUsers1': {
+        all: ['head-trainer'],
+        read : ['trainee', 'trainer'],
+        write : ['trainer'],
+        delete: [],
+        }
+};
+import{Ipermission} from './interfaces';
```
Comment (faiyaz24) [inline extraTs/constants.ts:17]: Import should be on top of the file.

File: extraTs/index.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    traineeEmail: "manish.dixit@successive.tech",
+    reviewerEmail: "faiyaz.ahmad@successive.tech"
+    },
+    {
+        traineeEmail: "manish.dixit@successive.tech98",
+       reviewerEmail: "faiyaz.ahmad@successive.tech"
+       }
+];
+
+import{Iusers} from './interfaces';
```
Comment (faiyaz24) [inline extraTs/index.ts:62]: Import on top.

File: extra/index.js (hunk 1 above)
Comment (manishdixit78) [reply extra/index.js:65]: ok sir

File: extra/utils/helpers.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+
+// Move the function validateEmail from validation.js to here
+let checkemail = /^[a-zA-Z0-9._-]+@[successive]+\.[tech]{2,4}$/;
```
Comment (faiyaz24) [inline extra/utils/helpers.js:3]: Checkemail can be defined as **Const**

File: extra/patterns/index.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import {diamondpattern} from "./diamond";
```
Comment (faiyaz24) [inline extra/patterns/index.js:1]: export * from './daimond'

File: extraTs/utils/validation.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+console.log("The users are: ",valid_users);
+let invalidlen:number=invalid_users.length;
+console.log('Total invalid users',invalidlen);
+console.log('The users is: ',invalid_users);
+}
+//validateUsers(users);
+
+
+//import function validateEmail from helpers.js file.
+import{validateEmail} from './helpers.js';
```
Comment (faiyaz24) [inline extraTs/utils/validation.ts:26]: import should be on top.

File: extraTs/utils/index.ts [hunk 7]
```diff
@@ -0,0 +1,10 @@
+
+// import the functions from permissions and validation files
+import {hasPermission} from "./permissions";
```
Comment (faiyaz24) [inline extraTs/utils/index.ts:3]: export * from './file'

Can follow same for all file if not exported default.

**Pull request merged**